app/pillar/handler/crud.go

```diff
@@ -101,3 +101,53 @@ func DeleteSearch(c *web.AppContext) {
 	err := service.DeleteSearch(c)
 	doRespond(c, nil, err)
 }
+
+func CreateUpdateForm(c *web.AppContext) {
```

Comment:
This function either creates or updates the form depending on what it's given.  It follows Pillar's Upsert policy.